### DISCUSSION TITLE
Read XML supplier invoices

### DISCUSSION THREAD

**Discussion by LukeinArket** [fattureincloud/api#461] [Q&A] [answered]
Hi All,
please could you sugget me what are the API to use to retrieve supplier invoices list and the XML related to them ?
Thanks.

**Comment by emajo** [comment, marked as answer]
Hi, I noticed you closed this discussion, and I apologize for the delayed response. I'm not sure if you still need an answer, but in case it might help others.
Supplier invoices in our API are called Received Documents, there's a method to list all of them ([list received documents](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/received_documents)) and you can find the related XML in the `attachments` field.

Let us know if you need further help,
thanks,
Emanuele.

**Reply by ctliv** [reply]
I have to retrieve a vendor invoice with both the PDF and the corresponding XML document (compliant with "Agenzia delle Entrate XSD fatture v1.2"). 

I looked for an `attachments` field  either in the [List Received Documents](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents) or in the [Get Received Document](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents/-document_id-) methods, but there is no such field. There is an `attachment_url` field that returns a temporary URL for the PDF document and an `attachment_preview_url` that does not seem related to the XML. 

AFAIK, an `attachments` field exists only in [listEmails](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/emails) method, which is not related to invoices. There is also a [getEInvoiceXML](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/issued_documents/-document_id-/e_invoice/xml) method, but it seems to apply only to issued e-invoices.

As it should be possible to retrieve the XML of a received invoice, could you please point me to the right field?

Thanks in advance.

**Comment by LukeinArket** [comment]
Ciao  Emanuele,
Grazie mille per la risposta. Alla fine ci sono arrivati da solo ma avevo
fretta ed ho chiesto al forum


Il mer 17 lug 2024, 12:39 Emanuele Saccomandi ***@***.***> ha
scritto:
